### PR TITLE
[pipelines] liquid staker hot fix

### PR DESCRIPTION
**Pull request by ochaloup**
Change priority for stake accounts taken for claiming into with liquid staker authority.

Some refactorings are part of the PR as considered some bad naming here around.

### CODE REVIEW

File: settlement-pipelines/src/stake_accounts.rs [hunk 1]
```diff
@@ -23,10 +23,13 @@ use validator_bonds_common::stake_accounts::{
 // TODO: better to be loaded from chain
 pub const STAKE_ACCOUNT_RENT_EXEMPTION: u64 = 2282880;
 
-// prioritize collected stake accounts by:
-// - 1. initialized, non-delegated
-// - 2. deactivating
-// - 3. any non-locked
+// 4bZ6o3eUUNXhKuqjdCnCoPAoLgWiuLYixKaxoa8PpiKk
+pub const MARINADE_LIQUID_STAKER_AUTHORITY: [u8; 32] = [
```
Comment (janlegner) [inline settlement-pipelines/src/stake_accounts.rs:27]: Please just use String, then Pubkey::from_str().unwrap()

File: settlement-pipelines/src/stake_accounts.rs [hunk 2]
```diff
@@ -92,22 +95,39 @@ pub fn get_stake_state_type(
...
+    } else {
+        Pubkey::default()
+    };
+    if staker == Pubkey::from(MARINADE_LIQUID_STAKER_AUTHORITY) {
+        match get_stake_state_type(stake_account, clock, stake_history) {
+            StakeAccountStateType::DelegatedAndActive => 0,
+            StakeAccountStateType::DelegatedAndActivating => 1,
+            StakeAccountStateType::DelegatedAndDeactivated => 2,
+            StakeAccountStateType::Initialized => 3,
+            StakeAccountStateType::DelegatedAndDeactivating => 4,
```
Comment (janlegner) [inline settlement-pipelines/src/stake_accounts.rs:119]:
```suggestion
            StakeAccountStateType::DelegatedAndDeactivating => 2,
            StakeAccountStateType::Initialized => 3,
            StakeAccountStateType::DelegatedAndDeactivated => 4,
```

File: settlement-pipelines/src/stake_accounts.rs (hunk 1 above)
Comment (ochaloup) [reply settlement-pipelines/src/stake_accounts.rs]: Ok (my thinking was to omit parsing).

**Comment by ochaloup** [comment]
@janlegner PR updated, can you please check again?

File: settlement-pipelines/src/stake_accounts.rs (hunk 1 above)
Comment (janlegner) [reply settlement-pipelines/src/stake_accounts.rs]: readability > computer time

**Review by janlegner — APPROVED** [review]

**Pull request merged**